File: Java/Day30.java
/*
Day 30: Implement Binary Search Algorithm 
*/

// Program
import java.util.Scanner;

public class Day30 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the number of elements: ");
        int n = sc.nextInt();

        int arr[] = new int[n];
        System.out.println("Enter the elements (space seperated): ");
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }

        System.out.println("Enter the element to be search: ");
        int k = sc.nextInt();

        System.out.println(String.format("The given element is present at %d index.", binarySearch(arr, n, k)));
    }

    public static int binarySearch(int[] arr, int n, int k) {
        int l = 0, h = n - 1;
        while (l <= h) {
            int mid = (l + h) / 2;
            if (arr[mid] == k) {
                return mid;
            } else if (arr[mid] > k) {
                h = mid - 1;
            } else {
                l = mid + 1;
            }
        }

        return -1;
    }
}

/*
 * 
 * Enter the number of elements:
 * 5
 * Enter the elements (space seperated):
 * 23 45 56 68 90
 * Enter the element to be search:
 * 56
 * The given element is present at 2 index.
 * 
 */

File: Python/Day30.py
""" 

Day 30:	Implement Binary Search Algorithm.

"""

#Program
def binary_search(list1, n):  
    low = 0  
    high = len(list1) - 1  
    mid = 0  
  
    while low <= high:  
        mid = (high + low) // 2  
  
        if list1[mid] < n:  
            low = mid + 1  
  
        elif list1[mid] > n:  
            high = mid - 1  
  
        else:  
            return mid  
  
    return -1  
  
  
m = int(input("Enter the number of elements to add in the list: ")) 
list1 = []
print("Enter the elements in ascending order \n")
for i in range (1, m+1):
    list1.append(int(input("Enter the number: ")))  


n = int(input("\nEnter the element to search: "))

print("\n")
result = binary_search(list1, n)  
  
if result != -1:  
    print("Element is present at index", str(result))  
else:  
    print("Element is not present in list1")  
    
    
    
    
""" 

Input and Output:

Enter the number: 12
Enter the number: 34
Enter the number: 56
Enter the number: 78
Enter the number: 888

Enter the element to search: 78


Element is present at index 3

"""